src/package.rs

```diff
@@ -250,33 +251,33 @@ impl Package {
   }
 }
 
+#[must_use]
 pub fn merge_refs(
   left: Refs,
   right: Refs,
-  alias: String,
+  alias: &str,
   with: Option<Vec<String>>,
 ) -> Refs {
-  let mut refs = right;
-  match with {
-    Some(ns) => {
-      let set: HashSet<String> = ns.iter().collect();
-      refs.retain(|k, _| set.contains(k));
-      if alias != String::from("") {
+  left.union_with(with.map_or_else(|| {
```

Comment:
There's a general stylistic pattern I'm trying to follow in this codebase of preferring `match` to `.map_or_else()` etc. on `Option` and `Result` types, especially when the contents of the the different branches are big.

In this particular case I definitely feel like `match` is more legible

Reply:
I think it is more idiomatic to use `map_or_else` but I will definitely change it if it is a stylistic concern.

Reply:
Yeah, I recognize that that's the Rust idiom here, but I want to try to keep the codebase as legible as possible to FP people (including me!) that know pattern matching but perhaps not all the semantics of the various Rust std functions for dealing with Option, Result, etc.